relayer/src/chain/cosmos/types/gas.rs

```diff
@@ -0,0 +1,71 @@
+use ibc_proto::cosmos::tx::v1beta1::Fee;
+
+use crate::chain::cosmos::calculate_fee;
+use crate::config::{ChainConfig, GasPrice};
+
+/// Default gas limit when submitting a transaction.
+const DEFAULT_MAX_GAS: u64 = 400_000;
+
+/// Fraction of the estimated gas to add to the estimated gas amount when submitting a transaction.
+const DEFAULT_GAS_PRICE_ADJUSTMENT: f64 = 0.1;
+
+const DEFAULT_FEE_GRANTER: &str = "";
+
+pub struct GasConfig {
+    pub default_gas: u64,
+    pub max_gas: u64,
+    pub gas_adjustment: f64,
+    pub gas_price: GasPrice,
+    pub max_fee: Fee,
+    pub fee_granter: String,
+}
+
+impl GasConfig {
+    pub fn from_chain_config(config: &ChainConfig) -> GasConfig {
```

Comment:
This is newly added to pre-compute all gas parameters so that we don't need to re-compute it over and over again.

Reply:
Great idea!